aeron-driver/src/main/java/io/aeron/driver/NetworkPublication.java

```diff
@@ -714,9 +726,12 @@ boolean hasSpies()
 
     void updateHasReceivers(final long timeNs)
     {
-        if (((timeOfLastStatusMessageNs + connectionTimeoutNs) - timeNs < 0) && hasReceivers)
+        livenessTracker.onIdle(timeNs, connectionTimeoutNs);
+        final boolean isLive = livenessTracker.hasReceivers();
+
+        if (hasReceivers != isLive)
```

Comment:
Why conditional assignment, i.e. volatile write?

Reply:
Yes, it is to prevent the cost of the volatile write when not required.